src/builtins.c

```diff
@@ -1005,7 +1005,9 @@ JL_CALLABLE(jl_f_invoke)
     jl_value_t *argtypes = args[1];
     JL_GC_PUSH1(&argtypes);
     if (jl_is_tuple(args[1])) {
-        // TODO: maybe deprecation warning, better checking
+        jl_depwarn("`invoke(f, (types...), ...)` is deprecated, "
```

Comment:
I'll probably only rebase after https://github.com/JuliaLang/julia/pull/18468 and finishing the TODO's so there's time for people feeling strongly either way to comment.